### PR TITLE
add k8s semconv migration non-normative doc

### PR DESCRIPTION
**Pull request by ChrsMark**
Part of https://github.com/open-telemetry/semantic-conventions/issues/1537

## Changes

This PR adds the non-normative k8s semconv migration docs described by https://github.com/open-telemetry/semantic-conventions/issues/1537.

So far it only includes changes from https://github.com/open-telemetry/semantic-conventions/pull/1427 but the goal is to keep adding content in this file as we go.

This PR is blocked until https://github.com/open-telemetry/semantic-conventions/pull/1427 is merged.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

/cc @TylerHelmuth

### CODE REVIEW

File: docs/non-normative/k8s-migration.md
Comment (ChrsMark) [inline docs/non-normative/k8s-migration.md:24]: We need to figure out how this will be applied on the Collector.

File: docs/non-normative/k8s-migration.md
Comment (ChrsMark) [reply docs/non-normative/k8s-migration.md:36]: That would be nice! One open question here is if these should be per component/receiver or the same across all the components.

I would go with sth like `semconv.k8s.stable` and `semconv.k8s.legacy` but would love to get more suggestions here. (or `<receiver_name>.semconv.k8s.stable` if we want to include the receiver name)

Comment (mx-psi) [reply docs/non-normative/k8s-migration.md:36]: I feel like they should apply to all components at the same time (at least that's what I would expect as an end user). Those names sound reasonable to me.

**Review by dmitryax — APPROVED** [review]

File: docs/non-normative/k8s-migration.md
Comment (TylerHelmuth) [inline docs/non-normative/k8s-migration.md:37]: similar to `k8s.dup`, will users be allows to set both to true to emit both semantics?

Comment (ChrsMark) [reply docs/non-normative/k8s-migration.md]: Yes, I applied a change to the feature gates' names and enhanced the description to make it clear.

The idea is:

```
- On alpha the old schema is enabled by default (`semconv.k8s.disableLegacy` defaults to false), 
  while the new schema is disabled (`semconv.k8s.enableStable` defaults to false).
- On beta/stable the old schema is disabled (`semconv.k8s.disableLegacy` defaults to true),
  while the new is enabled (`semconv.k8s.enableStable` defaults to true).
```

To have both schemas (`k8s/dup`) we need to configure:
- On alpha: `otelcol --config=config.yaml --feature-gates=semconv.k8s.enableStable`
- On Beta:  `otelcol --config=config.yaml --feature-gates=semconv.k8s.disableLegacy`

@TylerHelmuth @mx-psi would that make sense? (I'm always confused with the feature gates' life-cycle, please correct me here if I'm wrong :))

Comment (mx-psi) [reply docs/non-normative/k8s-migration.md]: To enable both the old and new set of metrics you would have to pass 
```
--config=config.yaml --feature-gates=-semconv.disableLegacy,+semconv.k8s.enableStable
```

Both feature gates would graduate from alpha to beta on the same release, and from beta to stable at the same time as well.

Comment (ChrsMark) [reply docs/non-normative/k8s-migration.md]: Clarified that in place.

**Comment by ChrsMark** [comment]
[v1.29.0](https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.29.0) release is out (this includes the changes this current doc references).

@TylerHelmuth @mx-psi shall we proceed with this or is there anything else missing here?

File: docs/non-normative/k8s-migration.md
Comment (TylerHelmuth) [inline docs/non-normative/k8s-migration.md:40]: Is there any way to make the gate names both "positive"? I think it will be clearer to users to do `--feature-gates=+semconv.k8s.enableLegacy,+semconv.k8s.enableStable` if you want both.

Comment (TylerHelmuth) [inline docs/non-normative/k8s-migration.md:40]:
```suggestion
- Both schemas can be enabled with `--feature-gates=-semconv.k8s.disableLegacy,+semconv.k8s.enableStable`.
```

Comment (ChrsMark) [reply docs/non-normative/k8s-migration.md:40]: Won't the `semconv.k8s.enableLegacy` default to false/disabled on `alpha` 🤔 ?

Comment (TylerHelmuth) [reply docs/non-normative/k8s-migration.md:40]: It would have to start at beta, and then be moved to alpha following these [lifecycle steps](https://github.com/open-telemetry/opentelemetry-collector/tree/main/featuregate#feature-lifecycle). It wouldn't be the standard way for us to use the gate.

If we want to go from alpha -> beta -> stable we have to use the negative name.

Comment (ChrsMark) [reply docs/non-normative/k8s-migration.md:40]: Ah I see now! I would be fine by this.

If we conclude with this approach it should be the same in the `hostmetrics` receiver and System SemConv as well (with the respective feature gates pair).
@mx-psi @dmitryax @braydonk @roger what do think?

File: docs/non-normative/k8s-migration.md
Comment (ChrsMark) [reply docs/non-normative/k8s-migration.md]: I think we have moved on here. Resolving this.

File: docs/non-normative/k8s-migration.md
Comment (ChrsMark) [reply docs/non-normative/k8s-migration.md]: Resolving this for now. We can continue at https://github.com/open-telemetry/semantic-conventions/pull/1597#discussion_r1866107784

**Comment by ChrsMark** [comment]
It would be helpful to have this soon so we can use it as base for listing the upcoming changes, i.e. in https://github.com/open-telemetry/semantic-conventions/pull/1636.

The only leftover here seems to be the names of the feature gates: https://github.com/open-telemetry/semantic-conventions/pull/1597#discussion_r1866107784

File: docs/non-normative/k8s-migration.md
Comment (TylerHelmuth) [reply docs/non-normative/k8s-migration.md]: Thinking about this more following the standard feature-gate processor is probably better.  As long as we document how to emit both users will have a solution.

**Review by TylerHelmuth — APPROVED** [review]

**Review by AlexanderWert — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by rogercoll — APPROVED** [review]

**Comment by ChrsMark** [comment]
FYI @open-telemetry/specs-semconv-maintainers we have a lot of approvals already here.

**Pull request merged**
